src/core/models.py

```diff
@@ -651,6 +650,27 @@ def self_article_path(self):
         if self.article_id:
             return os.path.join(settings.BASE_DIR, 'files', 'articles', str(self.article_id), str(self.uuid_filename))
 
+    def url(self):
+        from core.middleware import GlobalRequestMiddleware
+        request = GlobalRequestMiddleware.get_current_request()
+        url_kwargs = {'file_id': self.pk}
+
+        if request.journal and self.article_id:
+            return reverse(
+                'download_journal_file',
```

Comment:
Does this work? The docs suggest the argument "viewname can be a URL pattern name or the callable view object"
So `journal_views.download_journal_file` or `"journal_file"`?

Reply:
Raising NotImplementedError for these for the time being,